assertk/src/commonMain/kotlin/assertk/assert.kt

```diff
@@ -174,21 +174,6 @@ fun <T> Assert<T>.all(body: Assert<T>.() -> Unit) {
     }
 }
 
-/**
- * Asserts on the given block returning an `Assert<Result<T>>`. You can test that it returns a value or throws an exception.
- *
- * ```
- * assertThat { 1 + 1 }.isSuccess().isPositive()
- *
- * assertThat {
- *   throw Exception("error")
- * }.isFailure().hasMessage("error")
- * ```
- *
- * @see assertFailure
- */
-inline fun <T> assertThat(f: () -> T): Assert<Result<T>> = assertThat(runCatching(f))
```

Comment:
I know that it doesn't help the ide annoyance but I think this should go through one round of deprecation before removal. Would also allow `ReplaceWith` for easier updates.

Reply:
Updated.

Kotlin has an internal `@LowPriorityInOverload` annotation, but sadly it seems to only be available for the JVM and JS targets so it wouldn't help common code, sadly.

Reply:
Deprecation already fixes the sorting

![Screen Shot 2023-05-12 at 9 10 27 PM](https://github.com/willowtreeapps/assertk/assets/66577/3186af09-08fd-4932-9f85-928226fc9e2e)